File: apps/mobile/lib/data/models/device.dart
class Device {
  final String id;
  final String deviceCode;
  final String name;
  final String? category;
  final String? location;
  final String hourlyRate;
  final String depositAmount;
  final int freeMinutes;
  final String status;
  
  Device({
    required this.id,
    required this.deviceCode,
    required this.name,
    this.category,
    this.location,
    required this.hourlyRate,
    required this.depositAmount,
    required this.freeMinutes,
    required this.status,
  });
  
  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: json['id'] as String,
      deviceCode: json['device_code'] as String,
      name: json['name'] as String,
      category: json['category'] as String?,
      location: json['location'] as String?,
      hourlyRate: json['hourly_rate'] as String,
      depositAmount: json['deposit_amount'] as String,
      freeMinutes: json['free_minutes'] as int? ?? 5,
      status: json['status'] as String,
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'device_code': deviceCode,
      'name': name,
      'category': category,
      'location': location,
      'hourly_rate': hourlyRate,
      'deposit_amount': depositAmount,
      'free_minutes': freeMinutes,
      'status': status,
    };
  }
  
  bool get isAvailable => status == 'available';
}

File: services/api/app/db/models/device.py
"""设备模型：二维码业务码、位置、押金、免费分钟与小时单价。"""

import uuid
from datetime import datetime
from decimal import Decimal

from sqlalchemy import CHAR, DateTime, Integer, Numeric, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class Device(Base):
  __tablename__ = "devices"

  id: Mapped[str] = mapped_column(
    CHAR(36),
    primary_key=True,
    default=lambda: str(uuid.uuid4()),
  )
  device_code: Mapped[str] = mapped_column(String(32), unique=True, nullable=False, index=True)
  name: Mapped[str] = mapped_column(String(128), nullable=False)
  category: Mapped[str | None] = mapped_column(String(64), nullable=True, index=True)
  location: Mapped[str | None] = mapped_column(String(256), nullable=True)
  hourly_rate: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
  deposit_amount: Mapped[Decimal] = mapped_column(
    Numeric(12, 2),
    nullable=False,
    default=Decimal("0.00"),
    server_default="0.00",
  )
  free_minutes: Mapped[int] = mapped_column(Integer, nullable=False, default=5, server_default="5")
  status: Mapped[str] = mapped_column(
    String(16),
    nullable=False,
    default="available",
    server_default="available",
    index=True,
  )
  created_at: Mapped[datetime] = mapped_column(
    DateTime,
    nullable=False,
    server_default=func.now(),
  )
  updated_at: Mapped[datetime] = mapped_column(
    DateTime,
    nullable=False,
    server_default=func.now(),
    onupdate=func.now(),
  )
